### PR TITLE
week 5 initial practice tests

### PR DESCRIPTION
**Pull request by honeybeebaby**
first practice adding test code to a test.java file and fixing those failures.

### CODE REVIEW

File: walkthroughs/week-5-tdd/intro/src/main/java/com/google/sps/Greeter.java [hunk 1]
```diff
@@ -22,6 +22,9 @@
    * Returns a greeting for the given name.
    */
   public String greet(String name) {
-    return "Hello " + name;
+    String whitespaceRemoved = name.replaceAll(" ", "");
+    String fixedName = whitespaceRemoved.replaceAll("[^a-zA-Z0-9]", "");
```
Comment (ihsan314) [inline walkthroughs/week-5-tdd/intro/src/main/java/com/google/sps/Greeter.java]:
```suggestion
    String fixedName = name.replaceAll("[^a-zA-Z0-9]\\s*", "");
```
I believe this suggestion should work, but feel free to keep it as is.

**Review by ihsan314 — APPROVED** [review]
LGTM, though consider checking out my code suggestion.

File: walkthroughs/week-5-tdd/intro/src/main/java/com/google/sps/Greeter.java (hunk 1 above)
Comment (honeybeebaby) [reply walkthroughs/week-5-tdd/intro/src/main/java/com/google/sps/Greeter.java]: it did work! :)
thank you

**Review by ccondit — APPROVED** [review]

File: walkthroughs/week-5-tdd/intro/src/main/java/com/google/sps/Greeter.java [hunk 2]
```diff
@@ -22,6 +22,8 @@
    * Returns a greeting for the given name.
    */
   public String greet(String name) {
-    return "Hello " + name;
+    String fixedName = name.replaceAll("[^a-zA-Z0-9]\\s*", "");
+
```
Comment (kaveh096) [inline walkthroughs/week-5-tdd/intro/src/main/java/com/google/sps/Greeter.java]:
```suggestion
```
nit: extra whitespace?

**Review by kaveh096 — APPROVED** [review]

**Review by mmurray22 — APPROVED** [review]
LGTM

**Pull request merged**